components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -952,6 +996,31 @@ def getAnnotation(self, ns=None):
             return AnnotationWrapper._wrap(self._conn, rv[0].child, link=rv[0])
         return None
 
+    def getAnnotationCounts(self, ns=None):
+        """
+        Get the annotion counts for the current object
+        """
+
+        params = omero.sys.ParametersI()
+        params.addId(long(self._oid))
+
+        q = """
+            select al from %sAnnotationLink al
+            left outer join fetch al.child as an
+            where al.parent.id = :id
+            """ % self.OMERO_CLASS
+
+        if ns:
```

Comment:
Better to explicitly check here ```if ns is not None:``` since you may actually want to use an empty string etc which would evaluate to False.